tests/nativeapp/test_post_deploy_for_app.py

```diff
@@ -56,35 +57,49 @@ def test_sql_scripts(
     mock_cli_ctx,
     mock_execute_queries,
     mock_execute_query,
-    project_directory,
+    temp_dir,
 ):
     mock_conn.return_value = MockConnectionCtx()
     mock_cli_ctx.return_value = {
         "ctx": {"native_app": {"name": "myapp"}, "env": {"foo": "bar"}}
     }
-    with project_directory("napp_post_deploy") as project_dir:
-        processor = _get_run_processor(str(project_dir))
+    post_deploy_1 = dedent(
+        """\
+            -- app post-deploy script (1/2)
 
-        processor.execute_app_post_deploy_hooks()
+            select myapp;
+            select bar;
+            """
+    )
+    post_deploy_2 = "-- app post-deploy script (2/2)\n"
+
+    ProjectV11Factory(
```

Comment:
alright, I have to ask the uncomfortable question... This seems like a _lot_ of infrastructure to perform what is essentially a file write and a dictionary merge. Both of these things can be expressed with a lot less code than this factory-based approach. For the sake of argument, something like:

```py
pdf_res = create_project(version="1.0", pdf_append={
    "native_app": {
         "name": "myapp",
         "application": {
              "post_deploy": [...]
         }
    },
    "env": {
        "foo": "bar",
        "package_foo": "package_bar",
    }
})
```

what's the advantage of using the whole factory setup here, and having to extend it / maintain it in the future? To be clear, I'm not saying there is no value, I'm just not sure I see it yet, so I need your help to see it.

Reply:
had a chat about this offline. conclusion, factory boy has its merits in terms of composability and readability. 
Adding more fixtures and utils do not seem to have been adopted with low friction. hoping this structure could allow some uniform solution.